db/db_impl.cc

```diff
@@ -4953,12 +4974,21 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
       }
     }
   }
+
+  // Set current_memtble_info for memtable sealed callback
+  MemTableInfo memtable_info;
+  memtable_info.cf_name = cfd->GetName();
+  memtable_info.smallest_seqno = cfd->mem()->GetFirstSequenceNumber();
+
   cfd->mem()->SetNextLogNumber(logfile_number_);
   cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
   new_mem->Ref();
   cfd->SetMemtable(new_mem);
   context->superversions_to_free_.push_back(InstallSuperVersionAndScheduleWork(
       cfd, new_superversion, mutable_cf_options));
+
+  // Notify will drop and reacquire dbmutex
+  NotifyOnMemTableSealed(cfd, memtable_info);
```

Comment:
OTOH moving it up has an advantage as it avoids a race condition and makes sure that Flush callback is not delivered before Sealed callback is delivered. I will update the code. I will add comments to say that the code under lock should be non-failable.